### PR TITLE
Fix SDK data-plane error mapping + add langchain-superserve integration plan

### PR DESCRIPTION
**Pull request by mohamedsobhi777**
## Related Issues

N/A — surfaced while building the LangChain/LangGraph (Deep Agents) integration.

## Description

Fixes a bug in the Python SDK's `map_api_error`: it assumed the response body's `error` field is always a dict (`{"error": {"message": ...}}`, the control-plane shape), but the **data plane** returns a bare string (`{"error": "file not found"}`). As a result, `files.read` / `files.write` errors crashed with `AttributeError: 'str' object has no attribute 'get'` instead of raising `NotFoundError` — affecting every SDK consumer that hits a data-plane file error, not just the integration. The fix handles the string form (with a regression test). This PR also adds the research and implementation-plan docs under `spec/` for the standalone `langchain-superserve` Deep Agents integration that surfaced the bug; the integration package itself lives in its own repo.

## Additional Context

The SDK change alters shared error-mapping behavior, so it's worth a careful review. The `langchain-superserve` integration depends on this fix (pins `superserve>=0.7.7`); its full `SandboxIntegrationTests` suite passes against it on the `superserve/python-3.11` template.

## Testing

- [x] Tested locally
- [x] Added/updated tests
- [x] All tests passing

SDK unit suite: 235 passed (incl. the new `test_string_error_body_maps_to_typed_error`); ruff + mypy clean. The integration's full `SandboxIntegrationTests` suite (87 tests) was verified green live against the fixed SDK.

### CODE REVIEW

File: packages/python-sdk/src/superserve/errors.py [hunk 1]
```diff
@@ -111,8 +111,14 @@ def __init__(
...
-    message = error_data.get("message", f"API error ({status_code})")
-    code = error_data.get("code")
+    if isinstance(error_data, str):
+        # Data-plane (boxd) errors use a bare string: ``{"error": "..."}``.
+        message = error_data
+        code = None
+    else:
+        # Control-plane errors use ``{"error": {"message": ..., "code": ...}}``.
+        message = error_data.get("message", f"API error ({status_code})")
+        code = error_data.get("code")
```
[the remark this answers is not in the document]
Comment (mohamedsobhi777) [reply packages/python-sdk/src/superserve/errors.py]: Addressed in 57c0695. The mapper now branches on `isinstance(error_data, dict)` and stringifies any other non-dict shape via `str()`, so it no longer raises on a list/int/bool body (or the data-plane string). On the empty-string point: I kept the `API error (<status>)` fallback for falsy/None error values instead of emitting an empty message, so `{"error": ""}` still yields a useful message (the suggested `str(error_data)` would produce `""` there). Added `test_unexpected_error_type_does_not_raise` to lock it in.

**Review by awille-ai — APPROVED** [review]
## Summary

Approved. The production code change is narrowly scoped to SDK error mapping: it preserves existing control-plane error handling while adding support for data-plane `{"error": "..."}` responses and defensive handling for unexpected non-dict error shapes.

## What looks good

- Security: no new auth, tenant boundary, command execution, filesystem, or network surface is introduced by the SDK change. The docs include examples and plans only; I did not find leaked real credentials.
- Scalability: the change is constant-time local response parsing on an already-failed request path. No added I/O, retries, polling, or shared-resource pressure.
- Edge cases: the prior crash on string error bodies is fixed, and the follow-up bot concern around list/int/bool-style malformed bodies is addressed by type-checking before `.get()`.
- Tests: new regression coverage verifies a string data-plane error maps to `NotFoundError` and a malformed list error no longer crashes the mapper. CI is green for lint, format check, typecheck, tests, docs-nav check, and build.

## Key risks

No high or critical findings.

Low, non-blocking: the implementation intentionally maps falsy non-dict error values such as `{"error": ""}` or `{"error": false}` to the default `API error (<status>)` message. That is a reasonable behavior, but since it was discussed in the review thread, I would add an explicit unit test for the empty-string case to lock in the decision.

## Migration and rollout guidance

This is safe to roll out with the next Python SDK release. It improves an error path without changing successful request behavior. If the LangChain integration pins `superserve>=0.7.7`, make sure that SDK version includes this commit before publishing or tagging the integration.

## Recommendation

Proceed. No blocker.
